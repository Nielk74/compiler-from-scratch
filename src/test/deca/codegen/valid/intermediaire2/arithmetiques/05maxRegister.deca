// Description:
//    Programme réalisant des opérations arithmétiques utilisant 15 variables pour tester la bonne utilisation de la pile (RMAX = 15).
//
// Resultats:
//    04maxRegister.expected
//
// Historique:
//    cree le 09/06/2022

{
    int a = 1;
    int b = 1;
    int c = 1;
    int d = 1;
    int e = 1;
    int f = 1;
    int g = 1;
    int h = 1;
    int i = 1;
    int j = 1;
    int k = 1;
    int l = 1;
    int m = 1;
    int n = 1;
    int o = 3;
    float q;

    int p = (a + (b + (c + (d + (e + (f + (g + (h + (i + (j + (k + (l + (m + (n + o))))))))))))));
    
    println(p);

    p = (a - (b + (c + (d - (e + (f - (g + (h % (i + (j - k * (l - (m * (n + o)))))))))))));

    println(p);

    q = p - p / p - p / p + p / p - p / p + p / p - p / p + p / p - p / p + p / p - p / p / p;

    println(q);
}