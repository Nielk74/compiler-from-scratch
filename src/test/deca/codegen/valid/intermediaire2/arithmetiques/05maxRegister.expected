17
-2
-2.50000e+00
